### ISSUE TITLE
Scikit-learn's HDBSCAN Implementation

### ISSUE THREAD

**Issue by MaartenGr** [MaartenGr/BERTopic#2031]
In a recent version of scikit-learn, I believe it was [v1.3](https://scikit-learn.org/1.5/whats_new/v1.3.html#id8), HDBSCAN was implemented with base functionality. Considering scikit-learn is already a requirement of BERTopic it stands to reason to use that implementation instead of the [original implementation](https://github.com/scikit-learn-contrib/hdbscan) since scikit-learn has more contributors. Moreover, common installation issues related to HDBSCAN might be alleviated with this. 

There are a couple of issues worth mentioning:
* Calculation of probabilities is if I'm not mistaken, not implemented in scikit-learn's HDBSCAN 
  * A solution would be to use the cosine similarities as the default method of calculating probabilities
* The feature set is smaller than the original implementation
* Speed needs to be tested to identify whether this is worth it
* Accuracy, whatever that means in this context, might also need some exploration

For those reading this, I'm interested to hear what you all think about this suggested change!

**Comment by StarlightScribe** [comment]
Anything that simplifies the dependency structure is progress. Would it affect the use of the cuML implementation? If not, you could leave that as the faster alternative.

**Comment by MaartenGr** [comment]
It doesn't affect the cuML implementation since it should be nothing more than a drop-in replacement of the original implementation. One major disadvantage is that currently soft-clustering is not implemented in the scikit-learn version which makes it difficult to return probabilities.

**Comment by o-laurent** [comment]
@MaartenGr Could I provide a PR for this? Including `hdbscan` is indeed a problem for me as I strive to reduce the number of dependencies of my projects. In addition, in my experience, sklearn's HDBSCAN multiprocesses better and is therefore faster, but I could benchmark it.

EDIT: I might be wrong on the performance aspect.

**Comment by MaartenGr** [comment]
@o-laurent Not too long ago, I made some changes to the core code to remove HBDSCAN if necessary. You can see the documentation of that here: https://maartengr.github.io/BERTopic/getting_started/tips_and_tricks/tips_and_tricks.html#lightweight-installation

I might want to change the default HDBSCAN to sklearn's at some point but it doesn't have the same functionality yet (e.g., `approximate_predict`). Also, I'm not too sure if it creates the exact same clusters, so I would have to do additional quality checks.

**Comment by o-laurent** [comment]
Thank you so much @MaartenGr for the pointer!
